### PR TITLE
added endpoint to unpublish RT content removed from the feed

### PR DESCRIPTION
**Pull request by dragospitica**

### CODE REVIEW

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 1]
```diff
@@ -103,7 +110,9 @@ private ContentWriteController(
             LookupEntryStore lookupEntryStore,
             ContentResolver contentResolver,
             ContentWriter contentWriter,
-            EquivalenceBreaker equivalenceBreaker
+            EquivalenceBreaker equivalenceBreaker,
+            OldContentDeactivator oldContentDeactivator,
+            ApplicationConfigurationFetcher configurationFetcher
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: this is already passed in

File: src/main/java/org/atlasapi/query/QueryWebModule.java [hunk 2]
```diff
@@ -389,7 +391,9 @@ private ContentWriteController contentWriteController() {
                         contentResolver,
                         entryStore,
                         lookupWriter
-                )
+                ),
+                oldContentDeactivator,
+                configFetcher
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/query/QueryWebModule.java]: already passed in

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 3]
```diff
@@ -159,6 +174,44 @@ public WriteResponse unpublishContent(HttpServletRequest req, HttpServletRespons
...
+        }
+
+        String source = request.getParameter("source");
+        ImmutableList<String> valid_uris = ImmutableList.of(
+                request.getParameter("valid_uris")
+        );
+
+        Publisher publisher = Publisher.fromKey(source).requireValue();
+
+        oldContentDeactivator.deactivateOldContent(publisher, valid_uris, 50);
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: Pretty major: I don't see any checking that the api key has write permissions for this publisher.

Comment (dragospitica) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: @jamiesprax the API key validation happens above when fetching the config from the request. This logic was already there.

Comment (jamiesprax) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: @dragospitica are you sure? I'm pretty sure that just resolves an application for an API key. Check the other methods in this class near the end. They do appConfig.canWrite(<publisher> checks.

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 4]
```diff
@@ -159,6 +171,68 @@ public WriteResponse unpublishContent(HttpServletRequest req, HttpServletRespons
     }
 
     @Nullable
+    @RequestMapping(value = "/4/content.json", method = RequestMethod.DELETE)
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: /3.0/content.json
You're in Owl

**Pull request merged**